### PR TITLE
call SetNumThreads everytime to avoid missing omp thread setting

### LINKED ISSUE #17153: fc_op slows on multi-instance inference
When profile multi-instance inference on pyramid_dnn, we find fc_op slows on it.
### command:
- one instance:
```
./paddle/fluid/inference/tests/api/test_analyzer_pyramid_dnn --infer_model=third_party/inference_demo/pyramid_dnn/model/ --infer_data=third_party/inference_demo/pyramid_dnn/data.txt --gtest_filter=Analyzer_Pyramid_DNN.profile --paddle_num_threads=1 --repeat=10000 --zero_copy --warmup --num_threads=1
```
- two instance:
```
./paddle/fluid/inference/tests/api/test_analyzer_pyramid_dnn --infer_model=third_party/inference_demo/pyramid_dnn/model/ --infer_data=third_party/inference_demo/pyramid_dnn/data.txt --gtest_filter=Analyzer_Pyramid_DNN.profile --paddle_num_threads=1 --repeat=10000 --zero_copy --warmup --num_threads=2
```
### result (latency)

| | fc_op | remove fc_op, i.e use mul+add |
|---|---|---|
|1 threads|  0.13118 | 0.141566 |
|2 threads|0.291523   |0.152074  |

The remove fc_op codes are:
```
--- a/paddle/fluid/inference/tests/api/analyzer_pyramid_dnn_tester.cc
+++ b/paddle/fluid/inference/tests/api/analyzer_pyramid_dnn_tester.cc
@@ -110,6 +110,7 @@ void SetConfig(AnalysisConfig *cfg) {
   if (FLAGS_zero_copy) {
     cfg->SwitchUseFeedFetchOps(false);
   }
+  cfg->pass_builder()->DeletePass("fc_fuse_pass");
 }
```

### PR DESCRIPTION
**Pull request by LeoZhao-BJ**
resolve #17153
test=develop

### CODE REVIEW

**Review by luotao1 — APPROVED** [review]
LGTM! Thanks very much!

File: paddle/fluid/inference/api/analysis_predictor.cc [hunk 1]
```diff
@@ -192,9 +192,7 @@ void AnalysisPredictor::SetMkldnnThreadID(int tid) {
 bool AnalysisPredictor::Run(const std::vector<PaddleTensor> &inputs,
                             std::vector<PaddleTensor> *output_data,
                             int batch_size) {
-  if (UNLIKELY(config_.cpu_math_library_num_threads() > 1)) {
-    paddle::platform::SetNumThreads(config_.cpu_math_library_num_threads());
-  }
+  paddle::platform::SetNumThreads(config_.cpu_math_library_num_threads());
```
Comment (luotao1) [inline paddle/fluid/inference/api/analysis_predictor.cc:195]: Here the reason of remove `UNLIKELY(config_.cpu_math_library_num_threads() > 1)` is: https://github.com/PaddlePaddle/Paddle/issues/17153#issuecomment-489487740
I test this PR in pyramid_dnn, it doesn't impact the inference speedup in small model. @Superjomn

Comment (LeoZhao-BJ) [reply paddle/fluid/inference/api/analysis_predictor.cc:195]: omp_set_num_threads() just does nothing but set var, so the cost should be less.

**Review by Superjomn — APPROVED** [review]
LGTM

**Pull request merged**